kin-devplatform-sdk/src/main/java/kin/devplatform/balance/presenter/BalancePresenter.java

```diff
@@ -178,18 +187,12 @@ private void showWelcomeToKin() {
 
 	private void addObservers() {
 		orderRepository.addOrderObserver(orderObserver);
-		blockchainSource.addBalanceObserver(balanceObserver);
-	}
-
-	@Override
-	public void onDetach() {
-		super.onDetach();
-		removeObservers();
+		blockchainSource.addBalanceObserver(balanceObserver, true);
 	}
 
 	private void removeObservers() {
 		orderRepository.removeOrderObserver(orderObserver);
-		blockchainSource.removeBalanceObserver(balanceObserver);
+		blockchainSource.removeBalanceObserver(balanceObserver, true);
```

Comment:
Doesn't this will also make pressure on the blockchain? meaning that we are always try to be in connection?
And what if the app(current activity) is in onStop but still wants some service to keep listen and maybe log or do something else?
Maybe it will be good to do it only at onDestroy? although it is not promised that it will be called but probably the one that you are listen to have some timeout for listeners who are listening for too much time...

Reply:
The hosting app can still put pressure if it uses the balance observer and never releases it.
here we prevent a case where our code does that...
in our case, there's no need to listen outside the activity, and for overcoming "timeout" errors we're reconnecting at `onStart` in order to create a new SSE connection in case old one errored.